src/arch/xen/kernel_init.c

```diff
@@ -38,6 +42,12 @@ void xen_kernel_start(start_info_t * start_info) {
 	
 	init_events();
 
+
+	xenstore_init(start_info);
+	char list[32];
+	uint32_t len = xenstore_ls("/", list, 32);
+	printk("len = %d\n", len);
```

Comment:
you're calling printk before `diag` initialized, which performed later in `kernel_start`

Reply:
thx